File: Frontend/todo/src/models/todo.ts
export interface IRoot {}

export interface Todo extends IRoot {
  id: string;
  title: string;
  description?: string;
  deadline: Deadline;
  isDone: boolean;
  isOverdue: boolean;
  updatedAt?: Date;
}

interface Deadline {
  from: Date;
  to: Date;
}

File: Backend/TodoApp.Domain/Entities/Todo.cs
using Foundation.Core.Interfaces;
using Foundation.Core.Types;

namespace TodoApp.Domain.Entities
{
    public class Todo : BaseEntity, IAggregateRoot
    {
        public Todo()
        {
        }

        public string Title { get; set; }
        public string? Description { get; set; }
        public TimeFrame Deadline { get; set; }
        public bool IsDone { get; set; }
        public bool IsOverdue => !IsDone && DateTime.UtcNow > Deadline.To;
        public DateTime? UpdatedAt { get; set; }
    }
}
